### PR TITLE
[[ Bug 17870 ]] Improve speed of script formatting

### PR DESCRIPTION
**Pull request by montegoulding**

### CODE REVIEW

File: Toolset/palettes/script editor/behaviors/revseeditorbehavior.livecodescript [hunk 1]
```diff
@@ -1858,41 +1878,37 @@ end firstWordToEnd
...
-      put firstWordToEnd(tLine) after tResult
-      get textFormatLine(x, tResult)
+   local tPreviousLine = 0
+   local tTextLines
+   put pText into tTextLines
+   split tTextLines by return
+   
+   get textFormatLine(1, tTextLines, tPreviousLine)
+   put firstWordToEnd(tTextLines[1]) into tTextLines[1]
+   put firstWordToEnd(tTextLines[1]) into tResult
```
Comment (livecodeali) [inline Toolset/palettes/script editor/behaviors/revseeditorbehavior.livecodescript]: Should this just be 

```
put tTextLines[1] into tResult
```

?

Comment (montegoulding) [inline Toolset/palettes/script editor/behaviors/revseeditorbehavior.livecodescript]: That seems a reasonably logical presumption ;-)

**Comment by montegoulding** [comment]
Woops... I amended the wrong commit. Luckily @peter-b is on holidays 😈 

BTW While the prefs change may seem unrelated the script editor hammers the prefs in a repeat loop with every set causing a save so it's a general performance thing I found and resolved while profiling the script editor.

**Comment by livecodeali** [comment]
👍 
@livecode-vulcan review ok bc65de6

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok bc65de63f9d22f46633f9a145229608ee5cab899

**Comment by peter-b** [comment]
I've got my eye on you, Mr Goulding!

File: Toolset/palettes/script editor/behaviors/revseeditorbehavior.livecodescript [hunk 2]
```diff
@@ -958,6 +966,10 @@ command scriptUncomment
       exit scriptUncomment
    end if
    
+   local tOldLock
+   put the lockScreen into tOldLock
```
Comment (trevordevore) [inline Toolset/palettes/script editor/behaviors/revseeditorbehavior.livecodescript:970]: the logic for handling unlock screen is incorrect. lockscreen doesn't behave like lockmessages. As long as your lock/unlock calls are balanced then you are good to go. As the code is now, if the screen is locked already then you will end up adding unbalanced lock screens.

Comment (montegoulding) [inline Toolset/palettes/script editor/behaviors/revseeditorbehavior.livecodescript:970]: Thanks for spotting this @trevordevore

**Pull request merged**